### PR TITLE
Update log_file/level docs, add log_buffer_* parameters

### PR DESCRIPTION
**Pull request by domcleal**
I fixed the smart proxy default settings file over in https://github.com/theforeman/smart-proxy/pull/368 and included the same in the template here, so you may wish to wait until that's reviewed before merging this.

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -403,6 +411,8 @@
   validate_array($trusted_hosts)
   validate_re($log_level, '^(UNKNOWN|FATAL|ERROR|WARN|INFO|DEBUG)$')
   validate_re($plugin_version, '^(installed|present|latest|absent)$')
+  validate_integer($log_buffer, undef, 0)
```
Comment (ekohl) [inline manifests/init.pp]: What a weird API where the upper bound comes before the lower bound.

**Comment by ekohl** [comment]
:+1:

File: manifests/init.pp (hunk 1 above)
Comment (domcleal) [inline manifests/init.pp]: It caught me out!

File: templates/settings.yml.erb [hunk 2]
```diff
@@ -81,9 +81,14 @@
...
-# valid options are
-# WARN, DEBUG, Error, Fatal, INFO, UNKNOWN
+# Uncomment and modify if you want to change the log level
+# WARN, DEBUG, ERROR, FATAL, INFO, UNKNOWN
 :log_level: <%= scope.lookupvar("foreman_proxy::log_level") %>
+
+# Log buffer size and extra buffer size (for errors). Defaults to 3000 messages in total,
+# which is about 500 kB request.
+:log_buffer: <%= scope.lookupvar("foreman_proxy::log_buffer") %>
+:log_buffer_errors: <%= scope.lookupvar("foreman_proxy::log_buffer_errors") %>
```
Comment (lzap) [inline templates/settings.yml.erb]: I slightly decreased this before the patch was merged to 1500 in total which is 250 kB in size. I thought 0,5 MB is maybe too big for remote sites.

Comment (domcleal) [inline templates/settings.yml.erb]: Could you check https://github.com/theforeman/smart-proxy/pull/368 please (which has already been merged)?  The default in the code appears to be 2000 + 1000, and it was only the config file comment that differed.

Comment (lzap) [inline templates/settings.yml.erb]: Oh I just noticed your patch, okay.

Comment (lzap) [inline templates/settings.yml.erb]: Thanks for the catch, let's keep it that way. User's can configure....

**Comment by mmoll** [comment]
merged, thanks @domcleal!

**Pull request closed without merging**